pkg/virt-api/api.go

```diff
@@ -229,6 +230,7 @@ func (app *virtAPIApp) composeSubresources() {
 
 		restartRouteBuilder := subws.PUT(definitions.NamespacedResourcePath(subresourcesvmGVR)+definitions.SubResourcePath("restart")).
 			To(subresourceApp.RestartVMRequestHandler).
+			Consumes(mime.MIME_ANY).
```

Comment:
Shouldn't this be only json?

Reply:
I tried using `restful.MIME_JSON` but the server replies `415: Unsupported Media Type`.

I suspect it is due to `virtctl` not sending any `Content-Type` header.

Reply:
Quite strange, generally we set the ContentType:
https://github.com/kubevirt/kubevirt/blob/main/staging/src/kubevirt.io/client-go/kubecli/kubecli.go#L327-L332
I'd expect the problem to be only in some stream requests as `virtctl console`...

Reply:
Unfortunately it happens also with (at least) start operation.

```
> virtctl start cirros-container-disk

PUT /apis/subresources.kubevirt.io/v1/namespaces/default/virtualmachines/cirros-container-disk/start HTTP/1.1\r\n
Host: 127.0.0.1:8001\r\n
User-Agent: Go-http-client/1.1\r\n
Content-Length: 2\r\n
Accept: application/json, */*\r\n
Accept-Encoding: gzip\r\n
\r\n
{}

HTTP/1.1 415 Unsupported Media Type\r\n
Audit-Id: 9d709cc9-3d60-4c0d-a811-c1e2d17f9c38\r\n
Cache-Control: no-cache, private\r\n
Content-Length: 27\r\n
Content-Type: text/plain; charset=utf-8\r\n
Date: Fri, 17 Nov 2023 10:10:09 GMT\r\n
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload\r\n
X-Kubernetes-Pf-Flowschema-Uid: 55d2a2f8-a04f-465f-98db-5f9fd7b0b4d5\r\n
X-Kubernetes-Pf-Prioritylevel-Uid: 9fd36ad4-180c-4076-8853-0aced74c1420\r\n
\r\n
415: Unsupported Media Type
```

Reply:
:thinking: will have a look

Reply:
> https://github.com/kubevirt/kubevirt/blob/main/staging/src/kubevirt.io/client-go/kubecli/kubecli.go#L327-L332 

Looks like this is used to set the `Accept` header and not the `Content-type`